#18554 v1.10 backport

jibi opened the pull request:
* https://github.com/cilium/cilium/pull/18554 -- Enable CI for feature branches (@jibi)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 18554; do contrib/backporting/set-labels.py $pr done 1.10; done
```

jibi committed:
workflows: enable CI for feature branches

[ upstream commit c7d983637515c1010cc7db4d3c0ed75e8abb5a88 ]

This commit enables CI for all feature branches based on the v1.10 one.

The naming convention for the base feature branch is:

  ft/v1.10/<feature>

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>

kkourt commented on .github/workflows/lint-codeql.yaml:
```diff
@@ -4,6 +4,7 @@ on:
+      - ft/v1.10/**
   push:
     branches:
       - v1.10
```
@jibi :wave: I'm going over the v1.11 backport of the same patch, and was wondering whether we need to add a line under branches here as well?

i.e., 
```
    branches:
        - v1.10
        - ft/v1.10/**
```

in the case of 1.10.

jibi replied:
Yes, for v1.11 we'll need to change that to `ft/v1.11/**`

nbusseneau commented on Documentation/contributing/development/contributing_guide.rst:
```diff
@@ -15,6 +15,10 @@ Clone and Provision Environment
+#. Fork the Cilium repository to your GitHub user or organization.
+#. Turn off GitHub actions for your fork as described in the `GitHub Docs <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository>`_.
+   #. This is recommended to avoid unnecessary CI notification failures on the fork.
+#. Clone your ``${YOUR_GITHUB_USERNAME_OR_ORG}/cilium`` fork into your ``GOPATH``, and setup the base repository as ``upstream`` remote:
```
I'm not sure we should update the documentation for the backports 🤔

jibi replied:
Yeah, on a second thought I should probably not. Let me get rid of it

jibi committed:
workflows: allow CI to run on fork repositories

[ upstream commit cc85af87ac65afebf047d1d7a20a7ce2aa16c4d6 ]

Currently CI will run only on the base cilium/cilium repo, due to a
check in most workflows in the form of:

    if: ${{ github.repository == 'cilium/cilium' }}

The original intent of this check was to avoid running CI on forks,
since workflow secrets are usually not configured, resulting in multiple
failures.

It turned out that the ability to run CI on forks is something useful to
have, and as a bonus point it would ease the development and maintenance
of new and existing workflows.

Because of this, this commit removes the aforementioned check.

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>
